### PR TITLE
Update dependent libraries jan10

### PR DESCRIPTION
**Pull request by mikeckennedy**
Updates the remote libraries (jquery, bootstrap to latest minor versions).

Bootstrap 3.0.3 -> 3.3.7
Jquery 1.10.2 -> 1.12.4
respond.js 1.3.0 -> 1.4.2

### CODE REVIEW

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
+    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
 
     <!-- Custom styles for this scaffold -->
     <link href="{{ '{{request.static_url(' }}'{{cookiecutter.repo_name}}:static/theme.css'){{ '}}' }}" rel="stylesheet">
 
     <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
     <!--[if lt IE 9]>
       <script src="//oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js" integrity="sha384-0s5Pv64cNZJieYFkXYOTId2HMA2Lfb6q2nAcx2n0RTLUnCAoTTsS0nKEO27XyKcY" crossorigin="anonymous"></script>
-      <script src="//oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js" integrity="sha384-f1r2UzjsxZ9T4V1f2zBO/evUqSEOpeaUUZcMTz1Up63bl4ruYnFYeM+BxI4NhyI0" crossorigin="anonymous"></script>
+      <script src="//oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
```
Comment (mmerickel) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: It looks like all the respond.js links got integrity removed.. intentional?

Comment (mikeckennedy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: @mmerickel  No, I just can't find it on oss.maxcdn.com for some reason. I'd be happy to add it back if you know what the SHA is.

Comment (mmerickel) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: Can't you just download the file and pipe it through `shasum -a 384` ? Maybe I'm missing something.

**Comment by mikeckennedy** [comment]
Yeah. I’ll do it tomorrow  and update the pr

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 (hunk 1 above)
Comment (mmerickel) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: `66868c6c317ee0b785c60e96752710f679d1d500b63084710353bd2961174309ecd46f143722046484227736f44f57e8`

Comment (mmerickel) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: Sorry it's `sha384-ZoaMbDF+4LeFxg6WdScQ9nnR1QC2MIRxA1O9KWEXQwns1G8UNyIEZIQidzb0T1fo`.

Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: Confirmed using this tool from Mozilla, which also has the command to use:
https://www.srihash.org/

`openssl dgst -sha384 -binary FILENAME.js | openssl base64 -A`

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 [hunk 2]
```diff
@@ -11,15 +11,15 @@
     <title>Cookiecutter Starter project for the Pyramid Web Framework</title>
 
     <!-- Bootstrap core CSS -->
-    <link href="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/css/bootstrap.min.css" rel="stylesheet">
+    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:14]: Verified.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 [hunk 3]
```diff
@@ -58,7 +58,6 @@
     <!-- Bootstrap core JavaScript
     ================================================== -->
     <!-- Placed at the end of the document so the pages load faster -->
-    <script src="//oss.maxcdn.com/libs/jquery/1.10.2/jquery.min.js" integrity="sha384-aBL3Lzi6c9LNDGvpHkZrrm3ZVsIwohDD7CDozL0pk8FwCrfmV7H9w8j3L7ikEv6h" crossorigin="anonymous"></script>
-    <script src="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/js/bootstrap.min.js" integrity="sha384-s1ITto93iSMDxlp/79qhWHi+LsIi9Gx6yL+cOKDuymvihkfol83TYbLbOw+W/wv4" crossorigin="anonymous"></script>
-  </body>
+    <script src="//code.jquery.com/jquery-1.12.4.min.js" integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ=" crossorigin="anonymous"></script>
+    <script src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>  </body>
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: Verified integrity attribute.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 [hunk 4]
```diff
@@ -58,7 +58,6 @@
     <!-- Bootstrap core JavaScript
     ================================================== -->
     <!-- Placed at the end of the document so the pages load faster -->
-    <script src="//oss.maxcdn.com/libs/jquery/1.10.2/jquery.min.js" integrity="sha384-aBL3Lzi6c9LNDGvpHkZrrm3ZVsIwohDD7CDozL0pk8FwCrfmV7H9w8j3L7ikEv6h" crossorigin="anonymous"></script>
-    <script src="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/js/bootstrap.min.js" integrity="sha384-s1ITto93iSMDxlp/79qhWHi+LsIi9Gx6yL+cOKDuymvihkfol83TYbLbOw+W/wv4" crossorigin="anonymous"></script>
-  </body>
+    <script src="//code.jquery.com/jquery-1.12.4.min.js" integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ=" crossorigin="anonymous"></script>
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: Please use `integrity="sha384-nvAa0+6Qg9clwYCGGPpDQLVpLNn0fRaROjHqs13t4Ggj3Ez50XnGQqc/r8MhnRDZ"`

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.mako [hunk 5]
```diff
@@ -40,7 +40,7 @@
             <ul>
               <li><i class="glyphicon glyphicon-cog icon-muted"></i><a href="https://github.com/Pylons/pyramid">Github Project</a></li>
               <li><i class="glyphicon glyphicon-globe icon-muted"></i><a href="https://webchat.freenode.net/?channels=pyramid">IRC Channel</a></li>
-              <li><i class="glyphicon glyphicon-home icon-muted"></i><a href="https://pylonsproject.org">Pylons Project</a></li>
+              <li><i class="glyphicon glyphicon-home icon-muted"></i><a href="http://pylonsproject.org">Pylons Project</a></li>
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.mako]: Please revert.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.mako [hunk 6]
```diff
@@ -56,8 +56,7 @@
     <!-- Bootstrap core JavaScript
     ================================================== -->
     <!-- Placed at the end of the document so the pages load faster -->
-    <script src="//oss.maxcdn.com/libs/jquery/1.10.2/jquery.min.js" integrity="sha384-aBL3Lzi6c9LNDGvpHkZrrm3ZVsIwohDD7CDozL0pk8FwCrfmV7H9w8j3L7ikEv6h" crossorigin="anonymous"></script>
-    <script src="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/js/bootstrap.min.js" integrity="sha384-s1ITto93iSMDxlp/79qhWHi+LsIi9Gx6yL+cOKDuymvihkfol83TYbLbOw+W/wv4" crossorigin="anonymous"></script>
-  </body>
+    <script src="//code.jquery.com/jquery-1.12.4.min.js" integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ=" crossorigin="anonymous"></script>
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.mako]: See above comment.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.pt [hunk 7]
```diff
@@ -56,7 +57,7 @@
     <!-- Bootstrap core JavaScript
     ================================================== -->
     <!-- Placed at the end of the document so the pages load faster -->
-    <script src="//oss.maxcdn.com/libs/jquery/1.10.2/jquery.min.js" integrity="sha384-aBL3Lzi6c9LNDGvpHkZrrm3ZVsIwohDD7CDozL0pk8FwCrfmV7H9w8j3L7ikEv6h" crossorigin="anonymous"></script>
-    <script src="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/js/bootstrap.min.js" integrity="sha384-s1ITto93iSMDxlp/79qhWHi+LsIi9Gx6yL+cOKDuymvihkfol83TYbLbOw+W/wv4" crossorigin="anonymous"></script>
+    <script src="//code.jquery.com/jquery-1.12.4.min.js" integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ=" crossorigin="anonymous"></script>
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.pt:60]: See above comment.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 (hunk 3 above)
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: Please put `</body>` back on new line.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 (hunk 4 above)
Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2]: @mikeckennedy I think you missed one for jquery.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.mako (hunk 6 above)
Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.mako]: @mikeckennedy This one for jquery, too.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Looks like the instances of jquery got overlooked.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 [hunk 8]
```diff
@@ -58,7 +58,7 @@
     <!-- Bootstrap core JavaScript
     ================================================== -->
     <!-- Placed at the end of the document so the pages load faster -->
-    <script src="//oss.maxcdn.com/libs/jquery/1.10.2/jquery.min.js" integrity="sha384-aBL3Lzi6c9LNDGvpHkZrrm3ZVsIwohDD7CDozL0pk8FwCrfmV7H9w8j3L7ikEv6h" crossorigin="anonymous"></script>
-    <script src="//oss.maxcdn.com/libs/twitter-bootstrap/3.0.3/js/bootstrap.min.js" integrity="sha384-s1ITto93iSMDxlp/79qhWHi+LsIi9Gx6yL+cOKDuymvihkfol83TYbLbOw+W/wv4" crossorigin="anonymous"></script>
+    <script src="//code.jquery.com/jquery-1.12.4.min.js" integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ=" crossorigin="anonymous"></script>
```
Comment (stevepiercy) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:61]: One more instance of jquery, should be

`integrity="sha384-nvAa0+6Qg9clwYCGGPpDQLVpLNn0fRaROjHqs13t4Ggj3Ez50XnGQqc/r8MhnRDZ"`

**Review by stevepiercy — CHANGES_REQUESTED** [review]
one more jquery instance

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 (hunk 8 above)
Comment (mikeckennedy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:61]: Hi think the jquery is fine. They are using sha256, you are using sha384 so the hash will be different. But I got it from the CDN on [code.jquery.com](https://code.jquery.com/) directory.

```html
script
	src="https://code.jquery.com/jquery-1.12.4.min.js"
	integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ="
	crossorigin="anonymous"></script>
		
The integrity and crossorigin attributes are used for Subresource Integrity (SRI) checking. This allows browsers to ensure that resources hosted on third-party servers have not been tampered with. Use of SRI is recommended as a best-practice, whenever libraries are loaded from a third-party source. Read more at srihash.org 
```

Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:61]: I don't see the hash on https://code.jquery.com/ and the note you quoted says:

> Read more at [srihash.org](https://www.srihash.org/)

...which is from where I got the sha384 hash. Can you show where you got the sha256 hash? I can't verify it.

Comment (mikeckennedy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:61]: Hi Steve,

Sure, you have to pull up the link for jquery 1 min. Here's the screenshot and code I included above.

sha256

Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:61]: OK, and I forgot that I could just do this after downloading it:

`cat jquery-1.12.4.min.js | openssl dgst -sha256 -binary | openssl enc -base64 -A`

Double-verified!

**Review by stevepiercy — APPROVED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2 (hunk 8 above)
Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/layout.jinja2:61]: FTR, AdBlocker blocked that modal javascript for me, and I didn't notice it until just now.

**Comment by stevepiercy** [comment]
I think I will update the other 2 cookiecutters tomorrow with the changes from both #48 and #47, then run through the Pyramid docs. My OCD is twitching.

**Comment by mikeckennedy** [comment]
Thanks @stevepiercy it's great to have the new projects on latest goodness (at least minor version-wise).

**Pull request merged**